File: _experimental/_perl/numbers/lib/numbers_recursive.pl
#!/usr/bin/perl
use strict;
use warnings;

sub fibonacci {
    my ($N) = @_;
    if ($N <= 0) {
        return 0;
    } elsif ($N == 1) {
        return 1;
    } else {
        return fibonacci($N - 1) + fibonacci($N - 2);
    }
}

sub factorial {
    my ($N) = @_;
    if ($N <= 1) {
        return 1;
    } else {
        return $N * factorial($N - 1);
    }
}

sub sum_numbers {
    my ($N) = @_;
    if ($N <= 0) {
        return 0;
    } else {
        return $N + sum_numbers($N - 1);
    }
}

1; 

File: _experimental/_julia/numbers/src/NumbersRecursive.jl
#!/usr/bin/julia

module NumbersRecursive
export fibonacci, factorial_rec, sum_numbers

fibonacci(n::Int) :: Int = n <= 1 ? n : fibonacci(n - 1) + fibonacci(n - 2)
factorial_rec(n::Int) :: Int = n == 0 ? 1 : n * factorial_rec(n - 1)
sum_numbers(n::Int) :: Int = n == 0 ? 0 : n + sum_numbers(n - 1)
end